wasserstein/_wasserstein.py

```diff
@@ -28,9 +28,9 @@
 import wasserstein.config
 
 if wasserstein.config.openmp():
-	from ._wasserstein_omp import *
+	from .omp._wasserstein import *
 else:
-	from ._wasserstein_noomp import *
+	from .noomp._wasserstein import *
```

Comment:
Before this PR the installed directory structure is

```console
$ tree /home/feickert/.pyenv/versions/wasserstein-dev/lib/python3.11/site-packages/wasserstein
/home/feickert/.pyenv/versions/wasserstein-dev/lib/python3.11/site-packages/wasserstein
├── config.py
├── __init__.py
├── _wasserstein_omp.cpython-311-x86_64-linux-gnu.so
├── _wasserstein.py
└── wasserstein.py

0 directories, 5 files
```

and so

```console
$ python -c 'import wasserstein._wasserstein_omp; print(wasserstein._wasserstein_omp)'
<module 'wasserstein._wasserstein_omp' from '/home/feickert/Code/GitHub/forks/Wasserstein/wasserstein/_wasserstein_omp.cpython-311-x86_64-linux-gnu.so'>
```

was valid.

This PR introduces changes to the file structure such that

```console
 tree /home/feickert/.pyenv/versions/wasserstein-dev/lib/python3.11/site-packages/wasserstein
/home/feickert/.pyenv/versions/wasserstein-dev/lib/python3.11/site-packages/wasserstein
├── config.py
├── __init__.py
├── omp
│   ├── __init__.py
│   └── _wasserstein.cpython-311-x86_64-linux-gnu.so
├── _wasserstein.py
└── wasserstein.py

1 directory, 6 files
```

such that the import fails with `ModuleNotFoundError`:

```console
$ python -c 'import wasserstein.omp._wasserstein'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'wasserstein.omp._wasserstein'
```

also seen through

```console
 python -c 'import wasserstein._wasserstein'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/feickert/Code/GitHub/forks/Wasserstein/wasserstein/_wasserstein.py", line 31, in <module>
    from .omp._wasserstein import *
ModuleNotFoundError: No module named 'wasserstein.omp._wasserstein'
```

I think this might be related to the packaging discussions already had in https://github.com/j-s-ashley/Wasserstein/pull/5, but cc @j-s-ashley and @henryiii on this.

Reply:
Isn't the module named `wasserstein.omp._wasserstein` in your example file directory, though?

Reply:
And why are tests passing if there's an import error?

Reply:
> Isn't the module named `wasserstein.omp._wasserstein` in your example file directory, though?

@henryiii Yes, which is what I was showing with

```console
$ python -c 'import wasserstein.omp._wasserstein'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'wasserstein.omp._wasserstein'
```

unless I'm misunderstanding what you mean.

> And why are tests passing if there's an import error?

https://github.com/thaler-lab/Wasserstein/blob/b8afaea73a4ce9c2bcff16f760130e97561739a7/tests/test_config.py

Doesn't test the import process apparently.

Reply:
I think `_wasserstein.py` needs to get moved to under `omp/`. Also need to fix the

```
NameError: name 'cvar' is not defined
``` 

in it.

Reply:
No, it does not need to be moved, then you'd not be able to import the compiled extension.

Reply:
Ah yes.

Reply:
Ah, no everything is fine. What is happening is that I forgot that this doesn't use a `src` directory layout yet and so if your `cwd` is

```console
$ tree -L 1
.
├── binder
├── boost
├── CHANGELOG.md
├── CMakeLists.txt
├── demos
├── docs
├── examples
├── LICENSE
├── pyproject.toml
├── pytest.ini
├── README.md
├── scripts
├── tests
└── wasserstein

8 directories, 6 files
```

then

```console
$ uv pip install --upgrade .
$ python -c 'import wasserstein.omp._wasserstein; print (wasserstein.omp._wasserstein)'
```

will be importing from the local `wasserstein/` directory, which is obviously going to not work. However

```console
$ mkdir _debug
$ cd $_
$ python -c 'import wasserstein.omp._wasserstein; print (wasserstein.omp._wasserstein)'
<module 'wasserstein.omp._wasserstein' from '/home/feickert/.pyenv/versions/wasserstein-dev/lib/python3.11/site-packages/wasserstein/omp/_wasserstein.cpython-311-x86_64-linux-gnu.so'>
```

works as expected.